### ISSUE TITLE
Hot restart fails "Ignoring terminal input: "r" because we are busy"

### ISSUE THREAD

**Issue by otopba** [flutter/flutter#74559]
Hello!

Hot reload works only once on my flutter app. I don't know why. After first success hot reload next attempt is stuck.

If I use terminal, I see this:
Ignoring terminal input: "r" because we are busy

First attempt:
```
[+18841 ms] Skipping target: gen_localizations
[        ] complete
[   +2 ms] Performing hot reload...
[  +30 ms] Scanned through 885 files in 11ms
[        ] Syncing files to device ONEPLUS A6010...
[        ] Compiling dart to kernel with 1 updated files
[   +1 ms] <- recompile package:gohere/main.dart 46e73d2d-b917-4fda-b740-33eaf51e94cf
[        ] package:gohere/ui/page/edit_profile_page.dart
[        ] <- 46e73d2d-b917-4fda-b740-33eaf51e94cf
[ +335 ms] Updating files.
[ +182 ms] DevFS: Sync finished
[        ] Syncing files to device ONEPLUS A6010... (completed in 519ms)
[        ] Synced 0.0MB.
[ +539 ms] <- accept
[        ] reloaded 1 of 2205 libraries
[  +89 ms] Reassembling application
[+2003 ms] This is taking a long time; will now check for paused isolates.
[ +108 ms] Found 0 newly paused isolate(s).
```

Second attempt:

```
[+345732 ms] Ignoring terminal input: "r" because we are busy.
```

If I use Android Studio, I see infinity progress "performing hot reload"
![img-2021-01-23-15-20-45](https://user-images.githubusercontent.com/4225279/105578075-a92d7580-5d8e-11eb-8d83-c97577b9d781.png)

macOS Big Sur 11.1
OnePlus 6T Android 10.0
Flutter channel master, 1.26.0-13.0.pre.79
Dart 2.12.0

flutter -doctor --verbose
```
client git:(master) ✗ flutter doctor -v
[✓] Flutter (Channel master, 1.26.0-13.0.pre.79, on macOS 11.1 20C69 darwin-x64, locale en-RU)
    • Flutter version 1.26.0-13.0.pre.79 at /Users/yuriy.dorofeev/flutter
    • Framework revision c66596e595 (3 days ago), 2021-01-20 12:19:18 -0800
    • Engine revision 0c79393dde
    • Dart version 2.12.0 (build 2.12.0-240.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/yuriy.dorofeev/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Users/yuriy.dorofeev/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/yuriy.dorofeev/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • COL L29 (mobile)       • CLCDU18703001438 • android-arm64  • Android 10 (API 29)
    • ONEPLUS A6010 (mobile) • ae52d68a         • android-arm64  • Android 10 (API 29)
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 88.0.4324.96

• No issues found!
```

**Comment by pedromassangocode** [comment]
Hi @otopba 
I cannot reproduce this on latest master channel.

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 1.26.0-13.0.pre.144, on Mac OS X 10.15.7 19H114 darwin-x64, locale en)
    • Flutter version 1.26.0-13.0.pre.144 at /Users/pedromassango/Code/flutter_master
    • Framework revision 2007186d2e (11 hours ago), 2021-01-24 12:54:00 -0800
    • Engine revision fdddf87080
    • Dart version 2.12.0 (build 2.12.0-253.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 52.2.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • Mac OS X 10.15.7 19H114 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 87.0.4280.141

! Doctor found issues in 1 category.
```

</details>

**Comment by otopba** [comment]
> Hi @otopba
> I cannot reproduce this on latest master channel.
> 
> flutter doctor -v
> ```shell
> [✓] Flutter (Channel master, 1.26.0-13.0.pre.144, on Mac OS X 10.15.7 19H114 darwin-x64, locale en)
>     • Flutter version 1.26.0-13.0.pre.144 at /Users/pedromassango/Code/flutter_master
>     • Framework revision 2007186d2e (11 hours ago), 2021-01-24 12:54:00 -0800
>     • Engine revision fdddf87080
>     • Dart version 2.12.0 (build 2.12.0-253.0.dev)
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
>     • Android SDK at /Users/pedromassango/Library/Android/sdk
>     • Platform android-30, build-tools 30.0.2
>     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
>     • All Android licenses accepted.
> 
> [!] Xcode - develop for iOS and macOS
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Xcode 12.1, Build version 12A7403
>     ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
>         CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
>         Without CocoaPods, plugins will not work on iOS or macOS.
>         For more info, see https://flutter.dev/platform-plugins
>       To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] Android Studio (version 4.1)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
> 
> [✓] IntelliJ IDEA Community Edition (version 2020.3.1)
>     • IntelliJ at /Applications/IntelliJ IDEA CE.app
>     • Flutter plugin version 52.2.5
>     • Dart plugin version 203.6912
> 
> [✓] VS Code (version 1.52.1)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.18.1
> 
> [✓] Connected device (2 available)
>     • macOS (desktop) • macos  • darwin-x64     • Mac OS X 10.15.7 19H114 darwin-x64
>     • Chrome (web)    • chrome • web-javascript • Google Chrome 87.0.4280.141
> 
> ! Doctor found issues in 1 category.
> ```

This is not reproducible with all projects. In some of my projects it works, in some it doesn't. It looks like there is some dependence on the written code

**Comment by pedromassangocode** [comment]
Hi @otopba 
Can you please provide a [minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?
Thank you

**Comment by dolpheen** [comment]
Same here
Looks like after updating to master have problems with out big project. The "counter" works ok
No messages, just 'busy' after first hot-reload

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
